packages/@expo/prebuild-config/src/plugins/unversioned/expo-splash-screen/withAndroidSplashStrings.ts

```diff
@@ -4,14 +4,28 @@ import { getAndroidSplashConfig } from './getAndroidSplashConfig';
 
 const RESIZE_MODE_KEY = 'expo_splash_screen_resize_mode';
 const STATUS_BAR_TRANSLUCENT_KEY = 'expo_splash_screen_status_bar_translucent';
+const FADE_DURATION_MS_KEY = 'expo_splash_screen_fade_duration_ms';
 
-export const withAndroidSplashStrings: ConfigPlugin = (config) => {
+type ExtraProps = {
+  resizeMode?: string;
+  fadeDurationMs?: number;
+};
+
+const defaultResizeMode = 'contain';
+
+export const withAndroidSplashStrings: ConfigPlugin<ExtraProps> = (config, splash) => {
   return withStringsXml(config, (config) => {
     const splashConfig = getAndroidSplashConfig(config);
     if (splashConfig) {
-      const { resizeMode } = splashConfig;
+      const resizeMode = splash?.resizeMode || defaultResizeMode;
```

Comment:
```suggestion
      const resizeMode = splash?.resizeMode ?? defaultResizeMode;
```

Reply:
also feel confused for the code a little bit:

```ts
    const splashConfig = getAndroidSplashConfig(config);
    if (splashConfig) {
      const resizeMode = splash?.resizeMode ?? defaultResizeMode;
      const fadeDurationMs = `${splash?.fadeDurationMs}`;
   }
```

in this case, the default splash config is there, but we read the value from extra props?

maybe that's more like what you want?

```ts
export const withAndroidSplashStrings: ConfigPlugin<ExtraProps> = (config, splash) => {
  return withStringsXml(config, (config) => {
    const splashConfig = getAndroidSplashConfig(config);
    const statusBarTranslucent = !!config.androidStatusBar?.translucent;
    const resizeMode = splash?.resizeMode ?? splashConfig?.resizeMode ?? defaultResizeMode;
    const fadeDurationMs = splash?.fadeDurationMs; // expected the fadeDurationMs is not in the legacy splash config
    config.modResults = setSplashStrings(
      config.modResults,
      resizeMode,
      statusBarTranslucent,
      fadeDurationMs
    );
    return config;
  });
};
```